SSA/Projects/InstCombine/AliveHandwrittenLargeExamples.lean

```diff
@@ -1236,24 +1236,12 @@ def alive_simplifyMulDivRem290 (w : Nat) :
   simp_alive_ssa
   simp_alive_undef
   simp_alive_ops
-  intros A
-  rcases A with none | A  <;> simp [Option.bind, Bind.bind]
-  intros A
-  rcases A with none | A  <;> simp [Option.bind, Bind.bind]
-  by_cases h : w ≤ BitVec.toNat A
-  simp only [h]
-  simp
-  simp only [h]
-  simp
-  intros B
-  rcases B with none | B  <;> simp [Option.bind, Bind.bind]
-  by_cases h : w ≤ BitVec.toNat B
-  simp only [h]
-  simp
-  simp only [h]
-  simp
-  simp at h
-  simp [trsa h]
+  intros A B
+  rcases A with none | A  <;> simp [Option.bind, Bind.bind] <;>
+  rcases B with none | B  <;> simp [Option.bind, Bind.bind] <;>
```

Comment:
The pattern 

```lean
  rcases A with none | A  <;> simp [Option.bind, Bind.bind] <;>
```

is AFAIU a little dangerous, since it also runs full `simp` on the `.some` case, without actually closing the goal. I would suggest

```lean
  rcases A with none | A  <;> try (simp [Option.bind, Bind.bind]; done) <;>
```

This will perform the following:
- In the case of `none`, since `simp [Option.bind, Bind.bind]` closes the goal, `done` succeeds, and thus the `try(...)` keeps the goal closed.
- In the case of `some _`. `simp [Option.bind, Bind.bind]` **does not** close the goal, `done` fails, and thus `try` undoes the simplification from an unguarded `simp`, reverting to the original proof state.